components/library/src/taxonomies/mod.rs

```diff
@@ -70,22 +71,38 @@ impl TaxonomyItem {
             })
             .collect();
         let (mut pages, ignored_pages) = sort_pages_by_date(data);
-        let slug = slugify_paths(name, config.slugify.taxonomies);
+        let item_slug = slugify_paths(name, config.slugify.taxonomies);
+        let taxo_slug = slugify_paths(&taxonomy.name, config.slugify.taxonomies);
         let permalink = if taxonomy.lang != config.default_language {
-            config.make_permalink(&format!("/{}/{}/{}", taxonomy.lang, taxonomy.name, slug))
+            config.make_permalink(&format!("/{}/{}/{}", taxonomy.lang, taxo_slug, item_slug))
         } else {
-            config.make_permalink(&format!("/{}/{}", taxonomy.name, slug))
+            config.make_permalink(&format!("/{}/{}", taxo_slug, item_slug))
         };
+        //let permalink = if taxonomy.lang != config.default_language {
+        //config.make_permalink(&format!("/{}/{}/{}", taxonomy.lang, taxonomy.name, slug))
+        //} else {
+        //config.make_permalink(&format!("/{}/{}", taxonomy.name, slug))
+        //};
 
         // We still append pages without dates at the end
         pages.extend(ignored_pages);
 
-        TaxonomyItem { name: name.to_string(), permalink, slug, pages }
+        TaxonomyItem { name: name.to_string(), permalink, slug: item_slug, pages }
     }
 
     pub fn serialize<'a>(&'a self, library: &'a Library) -> SerializedTaxonomyItem<'a> {
         SerializedTaxonomyItem::from_item(self, library)
     }
+
+    pub fn merge(&mut self, other: &Self) {
```

Comment:
if you merge you can pass `Self` I think

Reply:
I thought this would cause ownership conflicts but after trying it out you are actually correct. Changed this in the next commit